server/mailet/integration-testing/src/test/java/org/apache/james/mailets/SizeGreaterThanIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+
+package org.apache.james.mailets;
+
+import static org.apache.james.mailets.configuration.CommonProcessors.ERROR_REPOSITORY;
+import static org.apache.james.mailets.configuration.Constants.DEFAULT_DOMAIN;
+import static org.apache.james.mailets.configuration.Constants.LOCALHOST_IP;
+import static org.apache.james.mailets.configuration.Constants.PASSWORD;
+import static org.apache.james.mailets.configuration.Constants.RECIPIENT;
+import static org.apache.james.mailets.configuration.Constants.awaitAtMostOneMinute;
+
+import org.apache.james.MemoryJamesServerMain;
+import org.apache.james.mailets.configuration.CommonProcessors;
+import org.apache.james.mailets.configuration.MailetConfiguration;
+import org.apache.james.mailets.configuration.MailetContainer;
+import org.apache.james.mailets.configuration.ProcessorConfiguration;
+import org.apache.james.modules.protocols.ImapGuiceProbe;
+import org.apache.james.modules.protocols.SmtpGuiceProbe;
+import org.apache.james.transport.mailets.ToRepository;
+import org.apache.james.transport.matchers.SizeGreaterThan;
+import org.apache.james.utils.DataProbeImpl;
+import org.apache.james.utils.MailRepositoryProbeImpl;
+import org.apache.james.utils.SMTPMessageSender;
+import org.apache.james.utils.TestIMAPClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class SizeGreaterThanIntegrationTest {
+    public static final String POSTMASTER = "postmaster@" + DEFAULT_DOMAIN;
+    public static final String SENDER = "sender@" + DEFAULT_DOMAIN;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @Rule
+    public TestIMAPClient testIMAPClient = new TestIMAPClient();
+    @Rule
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+
+    private TemporaryJamesServer jamesServer;
+
+    @Before
+    public void setup() throws Exception {
+        jamesServer = TemporaryJamesServer.builder()
+            .withBase(MemoryJamesServerMain.SMTP_AND_IMAP_MODULE)
+            .withMailetContainer(
+                generateMailetContainerConfiguration())
+            .build(temporaryFolder.newFolder());
+
+        jamesServer.getProbe(DataProbeImpl.class)
+            .fluent()
+            .addDomain(DEFAULT_DOMAIN)
+            .addUser(RECIPIENT, PASSWORD)
+            .addUser(SENDER, PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void sizeGreaterThanMatcherShouldBounceWhenSizeExceeded() throws Exception {
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .sendMessageWithHeaders(SENDER, RECIPIENT, "01234567\r\n".repeat(1025));
+
+        awaitAtMostOneMinute.until(() -> jamesServer.getProbe(MailRepositoryProbeImpl.class).getRepositoryMailCount(ERROR_REPOSITORY) == 1);
```

Comment:
having a mail in error seems different than checking we bounced, or I don't understand something?

Reply:
We can use another repository name to disambiguate this IMO

Reply:
From wikipedia:

> A bounce message or just "bounce" is an automated message from an email system, informing the sender of a previous message that message had not been delivered

How checking a repository can prove we sent a bounce message?

Reply:
So far, the "bouncing behavior" have nothing to do with the "SizeGreaterThan" behavior. @remk is correctly testing the matcher behavior, and we should expect nothing more as part of this test class.

The fact of bouncing is enpowered by combining the SizeGreaterThan matcher with a Bounce mailet. The right way to go would IMO to add the SizeGreaterThan + Bounce combinaison as part of default configuration and add an integration test on a real server.

I propose to write the following tests:

```
When Alice sends a 9 MB  email via JMAP to BOB
Then Bob receives it
And Alice never receives a bounce email
```

```
When Alice sends a 11 MB  email via JMAP to BOB
Then Alice receives a bounce message
And Bob Did not receive the message
```

If this is not generic enough to make it to the default configuration, then maybe we have to consider creating a new (separated) project for writing integration tests for James with OpenPaaS configuration.

Reply:
> So far, the "bouncing behavior" have nothing to do with the "SizeGreaterThan" behavior. @remk is correctly testing the matcher behavior, and we should expect nothing more as part of this test class.

Yet the test name is describing we expect a bounce.

Reply:
> The fact of bouncing is enpowered by combining the SizeGreaterThan matcher with a Bounce mailet. The right way to go would IMO to add the SizeGreaterThan + Bounce combinaison as part of default configuration and add an integration test on a real server.

:+1: